test/tests/functional/pbs_qrun.py

```diff
@@ -95,52 +95,51 @@ def test_invalid_host_val(self):
+        verify whether qrun hangs.
         """
         node = self.mom.shortname
-        time1 = int(time.time())
-        walltime = 1
+        err_msg = 'qrun: Not enough free nodes available'
+        joblist = []
         self.server.manager(MGR_CMD_SET, SCHED,
                             {'scheduling': 'False'})
-        err_msg = 'qrun: Not enough free nodes available'
-        while True:
-            self.server.manager(MGR_CMD_SET, NODE,
-                                {'resources_available.ncpus': 1}, id=node)
-
-            j1 = Job(TEST_USER)
-            j1.create_script('/bin/date')
-            a = {'Resource_List.walltime': walltime}
-            j1.set_attributes(a)
-            self.server.submit(j1)
-
-            walltime = walltime + 1
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1}, id=node)
+        for walltime in range(1, 501):
+            j = Job(TEST_USER)
+            j.create_script('/bin/date')
             a = {'Resource_List.walltime': walltime}
-            j1.set_attributes(a)
-            jid2 = self.server.submit(j1)
-            walltime = walltime + 1
-            time2 = int(time.time())
+            j.set_attributes(a)
+            jid = self.server.submit(j)
+            joblist.append(jid)
+        self.logger.info("Submitted 500 jobs with different walltime")
 
-            self.server.manager(MGR_CMD_SET, SCHED,
-                                {'scheduling': 'True'})
-            self.server.manager(MGR_CMD_SET, SCHED,
-                                {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        job = 10  # when we turn on scheduling, it scheduled few job.
+        while job < 501:
+            jid = joblist[job]
+            now = int(time.time())
             try:
-                self.server.runjob(jobid=jid2)
+                self.server.runjob(jobid=jid)
             except PbsRunError as e:
+                # sched cycle started by qrun and failed as no free nodes
                 if err_msg in e.msg[0]:
                     self.logger.info(
                         "No free nodes available. Checking for sched cycle")
                     try:
                         self.scheduler.log_match("Starting Scheduling Cycle",
-                                                 starttime=time2,
+                                                 starttime=now,
                                                  max_attempts=10)
                     except PtlLogMatchError:
                         self.assertFalse(True,
-                                         "Sched cycle not started.Qrun hangs")
+                                         "Sched cycle not started by qrun")
                     else:
                         continue
```

Comment:
I don't think you need this else clause.  There is nothing else before the end of the loop